picocontainer/src/main/java/cucumber/runtime/java/picocontainer/PicoFactory.java

```diff
@@ -36,6 +41,18 @@ public boolean addClass(Class<?> clazz) {
         return true;
     }
 
+    /**
+     * Allows subclasses to configure {@link org.picocontainer.PicoContainer} with custom adapters.
+     * <p>
+     * For example, subclasses can use this method to configure custom factory methods
+     * by passing a {@link org.picocontainer.injectors.FactoryInjector}.
+     *
+     * @since 2.0.0
+     */
+    public void addAdapter(ComponentAdapter<?> componentAdapter) {
+        adapters.add(componentAdapter);
```

Comment:
While I see the utility of being able to configure more of the pico container I don't think that adding methods PicoFactory to be used by extending classes is the correct way to do this. There is fairly wide range of potential customizations that be applied by using the PicoBuilder and MutablePicoContainer. Supporting all these by adding more methods would be cumbersome. Additionally most users would configure only a small part of the pico container and thus only use a small of these methods. We'd en up with a large code base with little utility that mostly duplicates the PicoBuilder and Container.

So I'd rather see that the creation of the MutablePicoContainer was extracted into two protected methods. One for the creation of the PicoBuilder with the minimal settings  and one for the creation of the MutablePicoContainer. Subclasses would then be able to override either method, call the parent method, apply their modifications such as adding components and return the result.